experiments/storm/src/main/scala/io/gearpump/experiments/storm/util/GraphBuilder.scala

```diff
@@ -36,118 +37,90 @@ import scala.collection.JavaConversions._
 object GraphBuilder {
   val COMPONENT_ID = "component_id"
   val COMPONENT_SPEC = "component_spec"
-}
 
-private[storm] class GraphBuilder {
-  import GraphBuilder._
+  import StormUtil._
 
-  def build(topology: StormTopology)(implicit system: ActorSystem): Graph[Processor[_ <: Task], _ <: Partitioner] = {
+  def build(topology: StormTopology)(implicit system: ActorSystem): (Graph[Processor[_ <: Task], _ <: Partitioner], Map[Integer, String]) = {
     val processorGraph = Graph.empty[Processor[Task], Partitioner]
+    var taskToComponent = Map.empty[Integer, String]
 
     val spouts = topology.get_spouts()
-    val spoutTasks = spouts.map { case (id, spec) =>
-      val parallelism = getParallelism(spec.get_common())
-      val processor = Processor[StormProducer](parallelism,
-        taskConf = UserConfig.empty
-          .withString(COMPONENT_ID, id)
-          .withValue(COMPONENT_SPEC, spec)
-      )
-      id -> processor
-    }
     val bolts = topology.get_bolts()
-    val boltTasks = bolts.map { case (id, spec) =>
-      val parallelism = getParallelism(spec.get_common())
-      val processor = Processor[StormProcessor](parallelism,
-        taskConf = UserConfig.empty
-          .withString(COMPONENT_ID, id)
-          .withValue(COMPONENT_SPEC, spec)
-      )
-      id -> processor
-    }
+    var taskId = 0
 
-    spoutTasks.foreach { case (sourceId, sourceProducer) =>
-      val sourceSpec = spouts.get(sourceId)
-      getTargets(sourceId, topology).foreach { case (streamId, targets) =>
-        val outFields = new Fields(sourceSpec.get_common().get_streams().get(streamId).get_output_fields())
-        targets.foreach { case (targetId, grouping) =>
-          boltTasks.get(targetId).foreach { targetProcessor =>
-            processorGraph.addEdge(sourceProducer, groupingToPartitioner(outFields, grouping), targetProcessor)
-          }
-        }
+    val spoutProcessors = spouts.map { case (id, spout) => id -> spoutToProcessor(id, spout) }.toMap
```

Comment:
Can you break down this function to smaller private ones to make the purpose explicit?